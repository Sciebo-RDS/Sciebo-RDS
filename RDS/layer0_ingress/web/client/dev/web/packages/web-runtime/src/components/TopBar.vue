<template>
  <header
    class="oc-topbar uk-flex uk-flex-middle uk-flex-wrap oc-border-b oc-p-s"
    :aria-label="$gettext('Top bar')"
  >
    <oc-grid gutter="medium" flex>
      <div class="uk-hidden@l">
        <oc-button
          appearance="raw"
          class="oc-m-s oc-app-navigation-toggle"
          :aria-label="$gettext('Open navigation menu')"
          @click="toggleAppNavigationVisibility"
        >
          <oc-icon name="menu" />
        </oc-button>
      </div>
      <search-bar v-if="!isSearchDisabled" />
    </oc-grid>
    <oc-grid flex gutter="small" class="uk-width-expand uk-flex-right oc-m-rm">
      <feedback-link v-if="isFeedbackLinkEnabled" />
      <notifications v-if="activeNotifications.length" />
      <applications-menu v-if="applicationsList.length > 0" :applications-list="applicationsList" />
      <user-menu
        :user-id="userId"
        :user-display-name="userDisplayName"
        :applications-list="applicationsList"
      />
    </oc-grid>
  </header>
</template>

<script>
import { mapGetters } from 'vuex'
import ApplicationsMenu from './ApplicationsMenu.vue'
import UserMenu from './UserMenu.vue'
import Notifications from './Notifications.vue'
import SearchBar from './SearchBar.vue'
import FeedbackLink from './FeedbackLink.vue'

export default {
  components: {
    FeedbackLink,
    Notifications,
    ApplicationsMenu,
    UserMenu,
    SearchBar
  },
  props: {
    userId: {
      type: String,
      required: false,
      default: ''
    },
    userDisplayName: {
      type: String,
      required: false,
      default: ''
    },
    applicationsList: {
      type: Array,
      required: false,
      default: () => []
    },
    hasAppNavigation: {
      type: Boolean,
      required: false,
      default: false
    },
    activeNotifications: {
      type: [Array, Boolean],
      required: false,
      default: () => []
    }
  },
  computed: {
    ...mapGetters(['configuration']),

    isSearchDisabled() {
      return (
        this.configuration.options.hideSearchBar === true || this.$route.meta.hideSearchBar === true
      )
    },

    isFeedbackLinkEnabled() {
      return !this.configuration.options.disableFeedbackLink
    }
  },
  methods: {
    toggleAppNavigationVisibility() {
      this.$emit('toggleAppNavigationVisibility')
    }
  }
}
</script>

<style scoped>
.topbar-current-extension-title {
  color: white;
}

.oc-topbar {
  height: 60px;
}
</style>
