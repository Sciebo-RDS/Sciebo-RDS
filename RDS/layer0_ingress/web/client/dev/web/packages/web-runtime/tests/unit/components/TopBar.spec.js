import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import TopBar from 'web-runtime/src/components/TopBar.vue'
import stubs from '../../../../../tests/unit/stubs'
import axe from '../../../../../tests/unit/helper/axe'

const localVue = createLocalVue()
localVue.use(Vuex)

const globalSearch = enabled => ({
  options: { hideSearchBar: !enabled }
})

const routeWithHiddenSearchBar = () => ({ meta: { hideSearchBar: true } })
const defaultRoute = () => ({
  meta: {
    /* default is empty */
  }
})

describe('Top Bar component', () => {
  describe('when search bar visible globally', () => {
    it('Displays search bar if not disabled for specified route', async () => {
      const wrapper = shallowMount(TopBar, {
        store: new Vuex.Store({
          getters: {
            configuration: () => globalSearch(true)
          }
        }),
        localVue,
        stubs,
        propsData: {
          userId: 'einstein',
          userDisplayName: 'Albert Einstein'
        },
        mocks: {
          $route: defaultRoute()
        }
      })

      await expect(axe(wrapper.element)).resolves.toHaveNoViolations()
      expect(wrapper.html().indexOf('search-bar-stub')).toBeGreaterThan(-1)
      expect(wrapper).toMatchSnapshot()
    })

    it('Hides the search bar if disabled for specified route', () => {
      const wrapper = shallowMount(TopBar, {
        store: new Vuex.Store({
          getters: {
            configuration: () => globalSearch(true)
          }
        }),
        localVue,
        stubs,
        propsData: {
          userId: 'einstein',
          userDisplayName: 'Albert Einstein'
        },
        mocks: {
          $route: routeWithHiddenSearchBar()
        }
      })

      expect(wrapper.html().indexOf('search-bar-stub')).toEqual(-1)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when search bar hidden globally', () => {
    it('Hides search bar even if not disabled for specified route', () => {
      const wrapper = shallowMount(TopBar, {
        store: new Vuex.Store({
          getters: {
            configuration: () => globalSearch(false)
          }
        }),
        localVue,
        stubs,
        propsData: {
          userId: 'einstein',
          userDisplayName: 'Albert Einstein'
        },
        mocks: {
          $route: defaultRoute()
        }
      })

      expect(wrapper.html().indexOf('search-bar-stub')).toEqual(-1)
      expect(wrapper).toMatchSnapshot()
    })

    it('Hides the search bar if disabled for specified route', () => {
      const wrapper = shallowMount(TopBar, {
        store: new Vuex.Store({
          getters: {
            configuration: () => globalSearch(false)
          }
        }),
        localVue,
        stubs,
        propsData: {
          userId: 'einstein',
          userDisplayName: 'Albert Einstein'
        },
        mocks: {
          $route: routeWithHiddenSearchBar()
        }
      })

      expect(wrapper.html().indexOf('search-bar-stub')).toEqual(-1)
      expect(wrapper).toMatchSnapshot()
    })
  })

  it('Displays applications menu', () => {
    const wrapper = shallowMount(TopBar, {
      store: new Vuex.Store({
        getters: {
          configuration: () => globalSearch(false)
        }
      }),
      localVue,
      stubs,
      propsData: {
        userId: 'einstein',
        userDisplayName: 'Albert Einstein',
        applicationsList: ['testApp']
      },
      mocks: {
        $route: defaultRoute()
      }
    })

    expect(wrapper.html().indexOf('applications-menu-stub')).toBeGreaterThan(-1)
    expect(wrapper).toMatchSnapshot()
  })

  it('Emits toggle of app navigation visibility', async () => {
    const wrapper = shallowMount(TopBar, {
      store: new Vuex.Store({
        getters: {
          configuration: () => globalSearch(true)
        }
      }),
      localVue,
      stubs,
      mocks: {
        $route: defaultRoute()
      }
    })

    wrapper.find('.oc-app-navigation-toggle').vm.$emit('click')
    await wrapper.vm.$nextTick()
    expect(wrapper.emitted().toggleAppNavigationVisibility).toBeTruthy()
  })
})
