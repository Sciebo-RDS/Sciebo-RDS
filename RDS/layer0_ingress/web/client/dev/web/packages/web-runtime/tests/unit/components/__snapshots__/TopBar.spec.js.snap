// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top Bar component Displays applications menu 1`] = `
<header aria-label="Top bar" class="oc-topbar uk-flex uk-flex-middle uk-flex-wrap oc-border-b oc-p-s">
  <oc-grid-stub gutter="medium" flex="">
    <div class="uk-hidden@l">
      <oc-button-stub appearance="raw" aria-label="Open navigation menu" class="oc-m-s oc-app-navigation-toggle">
        <oc-icon-stub name="menu"></oc-icon-stub>
      </oc-button-stub>
    </div>
    <!---->
  </oc-grid-stub>
  <oc-grid-stub flex="" gutter="small" class="uk-width-expand uk-flex-right oc-m-rm">
    <feedback-link-stub></feedback-link-stub>
    <!---->
    <applications-menu-stub applicationslist="testApp"></applications-menu-stub>
    <user-menu-stub userid="einstein" userdisplayname="Albert Einstein" applicationslist="testApp"></user-menu-stub>
  </oc-grid-stub>
</header>
`;

exports[`Top Bar component when search bar hidden globally Hides search bar even if not disabled for specified route 1`] = `
<header aria-label="Top bar" class="oc-topbar uk-flex uk-flex-middle uk-flex-wrap oc-border-b oc-p-s">
  <oc-grid-stub gutter="medium" flex="">
    <div class="uk-hidden@l">
      <oc-button-stub appearance="raw" aria-label="Open navigation menu" class="oc-m-s oc-app-navigation-toggle">
        <oc-icon-stub name="menu"></oc-icon-stub>
      </oc-button-stub>
    </div>
    <!---->
  </oc-grid-stub>
  <oc-grid-stub flex="" gutter="small" class="uk-width-expand uk-flex-right oc-m-rm">
    <feedback-link-stub></feedback-link-stub>
    <!---->
    <!---->
    <user-menu-stub userid="einstein" userdisplayname="Albert Einstein" applicationslist=""></user-menu-stub>
  </oc-grid-stub>
</header>
`;

exports[`Top Bar component when search bar hidden globally Hides the search bar if disabled for specified route 1`] = `
<header aria-label="Top bar" class="oc-topbar uk-flex uk-flex-middle uk-flex-wrap oc-border-b oc-p-s">
  <oc-grid-stub gutter="medium" flex="">
    <div class="uk-hidden@l">
      <oc-button-stub appearance="raw" aria-label="Open navigation menu" class="oc-m-s oc-app-navigation-toggle">
        <oc-icon-stub name="menu"></oc-icon-stub>
      </oc-button-stub>
    </div>
    <!---->
  </oc-grid-stub>
  <oc-grid-stub flex="" gutter="small" class="uk-width-expand uk-flex-right oc-m-rm">
    <feedback-link-stub></feedback-link-stub>
    <!---->
    <!---->
    <user-menu-stub userid="einstein" userdisplayname="Albert Einstein" applicationslist=""></user-menu-stub>
  </oc-grid-stub>
</header>
`;

exports[`Top Bar component when search bar visible globally Displays search bar if not disabled for specified route 1`] = `
<header aria-label="Top bar" class="oc-topbar uk-flex uk-flex-middle uk-flex-wrap oc-border-b oc-p-s">
  <oc-grid-stub gutter="medium" flex="">
    <div class="uk-hidden@l">
      <oc-button-stub appearance="raw" aria-label="Open navigation menu" class="oc-m-s oc-app-navigation-toggle">
        <oc-icon-stub name="menu"></oc-icon-stub>
      </oc-button-stub>
    </div>
    <search-bar-stub></search-bar-stub>
  </oc-grid-stub>
  <oc-grid-stub flex="" gutter="small" class="uk-width-expand uk-flex-right oc-m-rm">
    <feedback-link-stub></feedback-link-stub>
    <!---->
    <!---->
    <user-menu-stub userid="einstein" userdisplayname="Albert Einstein" applicationslist=""></user-menu-stub>
  </oc-grid-stub>
</header>
`;

exports[`Top Bar component when search bar visible globally Hides the search bar if disabled for specified route 1`] = `
<header aria-label="Top bar" class="oc-topbar uk-flex uk-flex-middle uk-flex-wrap oc-border-b oc-p-s">
  <oc-grid-stub gutter="medium" flex="">
    <div class="uk-hidden@l">
      <oc-button-stub appearance="raw" aria-label="Open navigation menu" class="oc-m-s oc-app-navigation-toggle">
        <oc-icon-stub name="menu"></oc-icon-stub>
      </oc-button-stub>
    </div>
    <!---->
  </oc-grid-stub>
  <oc-grid-stub flex="" gutter="small" class="uk-width-expand uk-flex-right oc-m-rm">
    <feedback-link-stub></feedback-link-stub>
    <!---->
    <!---->
    <user-menu-stub userid="einstein" userdisplayname="Albert Einstein" applicationslist=""></user-menu-stub>
  </oc-grid-stub>
</header>
`;
